cpython/patches/0009-Fix-iPad-detection-in-wasm-gc.patch

```diff
@@ -0,0 +1,34 @@
+From 738e25376b23611b6077c71f5ef55eefa66dd217 Mon Sep 17 00:00:00 2001
+From: ryanking13 <def6488@gmail.com>
+Date: Wed, 11 Jun 2025 08:54:37 +0000
+Subject: [PATCH 9/9] Fix iPad detection in wasm-gc
+
+Fixes iPad detection logic in wasm-gc, which was malfunctioning in recent iPadOS + safari.
+
+---
+ Python/emscripten_trampoline.c | 9 ++++++++-
+ 1 file changed, 8 insertions(+), 1 deletion(-)
+
+diff --git a/Python/emscripten_trampoline.c b/Python/emscripten_trampoline.c
+index cc5047d6bd..64315ad5e9 100644
+--- a/Python/emscripten_trampoline.c
++++ b/Python/emscripten_trampoline.c
+@@ -71,7 +71,14 @@ EM_JS(CountArgsFunc, _PyEM_GetCountArgsPtr, (), {
+ // )
+ 
+ function getPyEMCountArgsPtr() {
+-    let isIOS = globalThis.navigator && /iPad|iPhone|iPod/.test(navigator.platform);
++    // Starting with iPadOS 13, iPads might send a platform string that looks like a desktop Mac.
++    // To differentiate, we check if the platform is 'MacIntel' (common for Macs and newer iPads)
++    // AND if the device has multi-touch capabilities (navigator.maxTouchPoints > 1)
++    let isIOS = globalThis.navigator && (
++        /iPad|iPhone|iPod/.test(navigator.userAgent) ||
++        (navigator.platform === 'MacIntel' && typeof navigator.maxTouchPoints !== 'undefined' && navigator.maxTouchPoints > 1)
```

Comment:
FYI `undefined > 1` is always `false` so that a check for a *type* as *undefined* is bloat *but*, most importantly, a check that access the property *might* have side-effects ... it's a long story but if you really want to check the type, you are better off with `'maxTouchPoints' in navigator` instead of `typeof navigator.maxTouchPoints` as `in` operator signals the presence of either accessors or values but it doesn't reach these (lazy accessors can backfire, I have learned this [long time ago](https://webreflection.blogspot.com/2011/08/please-stop-reassigning-for-no-reason.html))

Reply:
the fact `in` produces `true` but the value is `undefined` won't invalidate what I've said because `undefined > 1` is always false but, logically speaking, the check should be `typeof navigator.maxTouchPoints === 'number' && navigator.maxTouchPoints > 1` if we want to be sure that `typeof` has a reason to exist at all ... anyway, it's a *nit* but the current check is both redundant and not clean in its intent ... glad it shipped though 🥳